core/syscall/syscall.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2024 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package syscall
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/holiman/uint256"
+)
+
+var sysTxContext = vm.TxContext{
+	Origin:   params.SystemAddress,
+	GasPrice: common.Big0,
+}
+
+func newBlockContext(header *types.Header) vm.BlockContext {
+	return vm.BlockContext{
+		CanTransfer: func(db vm.StateDB, addr common.Address, amount *uint256.Int) bool { return false },
+		Transfer:    func(vm.StateDB, common.Address, common.Address, *uint256.Int) {},
+		GetHash:     func(uint64) common.Hash { return common.Hash{} },
```

Comment:
The block context is incomplete. While it may suffice for EIP4788, it's likely insufficient for future needs.

Reply:
Ideally, we should reuse the block context for transaction processing, ensuring that all necessary environmental fields are reliably set.

Reply:
Yeah I was wondering if we could move `core/evm.go` somewhere else. I needed to recreate this to avoid import cycle!